### PR TITLE
Feature transaction detail & Fix "In Progress" activity

### PR DESCRIPTION
**Pull request by CS6**
Feature transaction detail #220 
fix Can show different icon to distinguish different "In Progress" activity #199

### CODE REVIEW

**Comment by seanwu1105** [comment]
@CS6 ,`npm run lint && mom run test-ci` locally to avoid creating blocking PR. Also, you can create a draft PR before fixing the linting and testing errors.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <mat-card class="transaction-card">
+      <mat-card-content class="row">
+        <mat-label> {{ details.asset.id }}</mat-label>
+        <button [class]="details.status" mat-stroked-button disableRipple>
+          {{ details.status }}
+        </button>
+      </mat-card-content>
+      <img [src]="details.asset.asset_file_thumbnail" />
+      <mat-card-content class="column">
+        <mat-label>From : {{ details.sender }}</mat-label>
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.html:19]: Missing translation.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <mat-label>To : {{ details.receiver_email }}</mat-label>
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.html:20]: Missing translation.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<mat-toolbar color="primary" *transloco="let t">
+  <button routerLink=".." routerDirection="back" mat-icon-button>
+    <mat-icon>arrow_back</mat-icon>
+  </button>
+  <span>{{ t('TransactionDetails') }}</span>
+</mat-toolbar>
+<div *transloco="let t" class="page-content">
+  <ng-container>
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.html:8]: Unnecessary `ng-container`.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    <h4 class="top-time">{{ details.created_at | date: 'short' }}</h4>
+    <mat-card class="transaction-card">
+      <mat-card-content class="row">
+        <mat-label> {{ details.asset.id }}</mat-label>
+        <button [class]="details.status" mat-stroked-button disableRipple>
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.html:13]: Currently, the type of `details` does not contain `status` property. Please extend the type of `details` from `Transaction`.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  selector: 'app-transaction-details',
+  templateUrl: './transaction-details.page.html',
+  styleUrls: ['./transaction-details.page.scss'],
+})
+export class TransactionDetailsPage implements OnInit {
+  details!: Transaction;
+  readonly status = Status;
+
+  constructor(
+    private readonly router: Router,
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.ts:18]: Unused variable.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { Transaction } from '../../../../services/publisher/numbers-storage/numbers-storage-api.service';
+
+@UntilDestroy({ checkProperties: true })
+@Component({
+  selector: 'app-transaction-details',
+  templateUrl: './transaction-details.page.html',
+  styleUrls: ['./transaction-details.page.scss'],
+})
+export class TransactionDetailsPage implements OnInit {
+  details!: Transaction;
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.ts:14]: Incorrect type for the template use.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(
+    private readonly router: Router,
+    private readonly route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.route.paramMap
+      .pipe(map(() => window.history.state))
+      .subscribe(details => {
+        this.details = details;
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.ts:26]: Set `this.details` directly in its declaration by using `Observable`. For example,

```typescript
details$ = this.route.paramMap.pipe(
  map(() => history.state)
)
```

File: src/app/pages/home/activity/transaction-details/transaction-details.page.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      });
+  }
+}
+
+enum Status {
+  InSendProgress = 'inSendProgress',
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.ts:32]: Ambiguous naming: `InSendProgress`.

File: src/assets/i18n/en-us.json [hunk 9]
```diff
@@ -75,6 +75,7 @@
   "delivered": "Delivered",
   "returned": "Returned",
   "inProgress": "In Progress",
+  "inSendProgress": "In Progress",
```
Comment (seanwu1105) [inline src/assets/i18n/en-us.json:78]: From users' aspect, there's no way to tell the difference between `inSendProgress` and `inProgress` status from the same text, which makes this change fails to fix issue #199.

File: src/assets/i18n/en-us.json [hunk 10]
```diff
@@ -99,5 +100,6 @@
     "invitationEmail": "Enter friend's email to send the invitation.",
     "clickingSignupToAgreePolicy": "By clicking Sign Up, you agree to our <a href=\"https://docs.google.com/document/d/1ZtD-E4-zcnh8DxQCSgR1Efhb9UooJMr7fb7bzPENd24/edit?usp=sharing\" target=\"_blank\">Terms</a> and <a href=\"https://docs.google.com/document/d/1m-LiyZ9ZDs4y9F_IKvY4ProLOYuEA_vIe_GfCSPNhwQ/edit?usp=sharing\" target=\"_blank\">Privacy Policy</a>.",
     "sendPostCaptureAlert": "After a PostCapture is sent, the ownership will be transferred to the selected friend. Are you sure?"
-  }
+  },
+  "TransactionDetails": "活動資訊"
```
Comment (seanwu1105) [inline src/assets/i18n/en-us.json:104]: Rename and sort this property for consistency.

File: src/assets/i18n/zh-tw.json [hunk 11]
```diff
@@ -74,6 +74,7 @@
   "accepted": "已接收",
   "delivered": "已送出",
   "inProgress": "等待中",
+  "inSendProgress": "等待中",
```
Comment (seanwu1105) [inline src/assets/i18n/zh-tw.json:77]: From users' aspect, there's no way to tell the difference between `inSendProgress` and `inProgress` status from the same text, which makes this change fails to fix issue #199.

File: src/assets/i18n/zh-tw.json [hunk 12]
```diff
@@ -99,5 +100,6 @@
     "invitationEmail": "輸入好友電子信箱發送邀請。",
     "clickingSignupToAgreePolicy": "點擊「註冊」即表示你同意我們的《<a href=\"https://docs.google.com/document/d/1ZtD-E4-zcnh8DxQCSgR1Efhb9UooJMr7fb7bzPENd24/edit?usp=sharing\" target=\"_blank\">服務條款</a>》和《<a href=\"https://docs.google.com/document/d/1m-LiyZ9ZDs4y9F_IKvY4ProLOYuEA_vIe_GfCSPNhwQ/edit?usp=sharing\" target=\"_blank\">資料政策</a>》。",
     "sendPostCaptureAlert": "PostCapture 送出後，它的所有權將轉移至選定的朋友。您確定嗎？"
-  }
+  },
+  "TransactionDetails": "活動資訊"
```
Comment (seanwu1105) [inline src/assets/i18n/zh-tw.json:104]: Rename and sort this property for consistency.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.ts [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  ngOnInit() {
+    this.route.paramMap
+      .pipe(map(() => window.history.state))
+      .subscribe(details => {
+        this.details = details;
+      });
+  }
+}
+
+enum Status {
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.ts:31]: Avoid code duplication. Import `Status` enumeration from `activity.page.ts` instead.

**Review by seanwu1105 — CHANGES_REQUESTED** [review]

**Review by seanwu1105 — CHANGES_REQUESTED** [review]
Please update the branch to `develop` before requesting review.

**Comment by CS6** [comment]
@seanwu1105  我沒有很懂，你是重構了嗎？我發現我好像都要重寫

File: src/app/pages/home/activity/transaction-details/transaction-details.page.html [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  </button>
+  <span>{{ t('transactionDetails') }}</span>
+</mat-toolbar>
+<div *transloco="let t" class="page-content">
+  <h4 class="top-time">{{ (details$ | async)?.created_at | date: 'short' }}</h4>
+  <mat-card class="transaction-card">
+    <mat-card-content class="row">
+      <mat-label> {{ (details$ | async)?.asset.id }}</mat-label>
+      <button
+        [class]="(details$ | async)?.status.__zone_symbol__value"
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.html:13]: Do not use private properties from Angular, such as `__zone_symbol__value`. You should get the data asynchronously or resolve it first as `status` is a `Promise`.

File: src/app/pages/home/activity/transaction-details/transaction-details.page.html [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        mat-stroked-button
+        disableRipple
+      >
+        {{ (details$ | async)?.status.__zone_symbol__value }}
```
Comment (seanwu1105) [inline src/app/pages/home/activity/transaction-details/transaction-details.page.html:17]: Do not use private properties from Angular, such as `__zone_symbol__value`. You should get the data asynchronously or resolve it first as `status` is a `Promise`.

Also, translate it.

**Review by seanwu1105 — CHANGES_REQUESTED** [review]

**Review by seanwu1105 — APPROVED** [review]

**Pull request merged**